log/value.go

```diff
@@ -39,6 +39,36 @@ func Timestamp(t func() time.Time) Valuer {
 	return func() interface{} { return t() }
 }
 
+// TimestampFormat returns a Valuer that produces a TimeFormat value from
+// layout and the time returned by t. Users will probably want to use
+// DefaultTimestamp or DefaultTimestampUTC.
+func TimestampFormat(t func() time.Time, layout string) Valuer {
+	return func() interface{} {
+		return TimeFormat{
+			Time:   t(),
+			Layout: layout,
+		}
+	}
+}
+
+// A TimeFormat represents an instant in time and a layout used when
+// marshaling to a text format.
+type TimeFormat struct {
+	Time   time.Time
+	Layout string
+}
+
+func (tf TimeFormat) String() string {
+	return tf.Time.Format(tf.Layout)
+}
+
+// MarshalText implements encoding.TextMarshaller.
+func (tf TimeFormat) MarshalText() (text []byte, err error) {
+	b := make([]byte, 0, len(tf.Layout)+10)
```

Comment:
I can add a brief comment, but I don't know what to name the constant? Do you have any suggestions?

For reference, the technique is a reduction of the standard library code for `time.Time.Format`, seen here: https://golang.org/src/time/format.go?s=14285:14327#L437. Unfortunately that code doesn't provide any naming help or even document why they picked 10.